### PR TITLE
doesn't popup error message when user cancels the main class selection.

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -189,6 +189,44 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
     private isEmptyArray(configItems: any): boolean {
         return !Array.isArray(configItems) || !configItems.length;
     }
+
+    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption> {
```
Comment (yaohaizh) [inline src/configurationProvider.ts]: Why the folder can be undefined?  It is strange to declare a type of undefined? If it is not existing, just null value.

Comment (testforstephen) [inline src/configurationProvider.ts]: Declare the return type to `Promise<IMainClassOption | undefined>`

File: src/configurationProvider.ts [hunk 2]
```diff
@@ -189,6 +189,44 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
...
+        }
+        const selection = pickItems.length > 1 ?
+            await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
+            : pickItems[0];
+        if (selection && selection.item) {
+            return selection.item;
+        } else {
+            vscode.window.showErrorMessage("Please specify the mainClass (e.g. [mymodule/]com.xyz.MainClass) in the launch.json.");
+            this.log("usageError", "Please specify the mainClass (e.g. [mymodule/]com.xyz.MainClass) in the launch.json.");
+            return undefined;
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Given that you have used `return ` in the previous side branch, use `return` directly here to keep code consistent.

File: src/configurationProvider.ts [hunk 3]
```diff
@@ -189,6 +189,44 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
...
+            return {
+                description: details,
+                label: name,
+                item,
+            };
+        }).sort ((a, b): number => {
+            return a.label > b.label ? 1 : -1;
+        });
+        if (pickItems.length === 0) {
+            // user gives up the selection, quit sliently
```
Comment (testforstephen) [inline src/configurationProvider.ts]: typos. `silently`.

File: src/configurationProvider.ts [hunk 4]
```diff
@@ -189,6 +189,44 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
...
```
Comment (yaohaizh) [inline src/configurationProvider.ts]: There is no items. User doesn't have the selection yet. Should remove this comment?

**Pull request merged**